### PR TITLE
Simplify dependency structure for easier integration of universal-router package

### PR DESCRIPTION
**Pull request by ewilz**
1. Use node modules where possible.
2. Eliminate necessity for integrators to have `remappings.txt`
3. Include instructions for setting up submodules

If possible I'd like to eliminate the remappings that remove `src` directory from submodules. This could clash with other integrators who don't remap (unless of course this is the most common pattern?) cc @marktoda
~Right now this is broken since Permit2 also uses this remapping~ Made a branch https://github.com/Uniswap/permit2/pull/160

Attempting at least a short term solution to this issue: https://github.com/Uniswap/universal-router/issues/186
feedback welcome!

EDIT: with this setup and instructions, I was able to successfully compile a project with universal-router dependencies

### CODE REVIEW

**Comment by marktoda** [comment]
> unless of course this is the most common pattern

i feel that it is, but I didn't realize hardhat doesn't support it: https://github.com/NomicFoundation/hardhat/issues/3231. So yeah I think best to remove from permit2 for now unfortunately

**Comment by ewilz** [comment]
> > unless of course this is the most common pattern
> 
> i feel that it is, but I didn't realize hardhat doesn't support it: [NomicFoundation/hardhat#3231](https://github.com/NomicFoundation/hardhat/issues/3231). So yeah I think best to remove from permit2 for now unfortunately

who am I to judge as the submodules n00b, but it seems overly complicated to me to be altering file structures for imports  🤷‍♀️

**Comment by marktoda** [comment]
Yea it's a builtin solc thing not a foundry thing but reasonable to just skip it if package intended to be used as a dep

File: .github/workflows/forge.yml [hunk 1]
```diff
@@ -22,6 +22,28 @@ jobs:
           submodules: recursive
           token: '${{ secrets.PERMIT_POST_KEY }}'
```
Comment (marktoda) [inline .github/workflows/forge.yml:23]: can delete now as its public no?

File: .github/workflows/forge.yml [hunk 2]
```diff
@@ -22,6 +22,28 @@ jobs:
...
+      - uses: actions/cache@v1
+        with:
+          path: ${{ steps.yarn-cache.outputs.dir }}
+          key: yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            yarn-
+
+      - name: Install dependencies
+        run: |
+          npm config set '//registry.npmjs.org/:_authToken' "${{ secrets.NPM_AUTH_TOKEN }}" \
```
Comment (marktoda) [inline .github/workflows/forge.yml:43]: do we need this? all packages should be public

**Review by marktoda — APPROVED** [review]
LGTM w/ couple workflow cleanup ideas

**Comment by ewilz** [comment]
changed permit2 to not remap the solmate file structure, or else every dependent has to do that too

**Pull request merged**